From dd133468dad5c69f698526f4e8bd58ed361841af Mon Sep 17 00:00:00 2001
Message-Id: <dd133468dad5c69f698526f4e8bd58ed361841af.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Wed, 24 Sep 2014 09:15:27 +0530
Subject: [PATCH 307/312] FOR_UPSTREAM [VPG]: drm/i915: Check presence of HDMI
 based on VBT.

Patch does HDMI initialization based on VBT.

Issue: GMIN-1509
Change-Id: Id20bfa94c6ce6f70c40a158d3643c0ae655770b1
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |   13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index c81a865..c5b8807 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11990,6 +11990,7 @@ static void intel_setup_outputs(struct drm_device *dev)
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	struct intel_encoder *encoder;
 	bool dpd_is_edp = false;
+	int i;
 
 	intel_lvds_init(dev);
 
@@ -12061,8 +12062,16 @@ static void intel_setup_outputs(struct drm_device *dev)
 			intel_dsi_init(dev);
 		else if (I915_READ(VLV_DISPLAY_BASE + DP_C) & DP_DETECTED)
 			intel_dp_init(dev, VLV_DISPLAY_BASE + DP_C, PORT_C);
-		if (I915_READ(VLV_DISPLAY_BASE + GEN4_HDMIB) & SDVO_DETECTED) {
-			intel_hdmi_init(dev, VLV_DISPLAY_BASE + GEN4_HDMIB, PORT_B);
+		/* Check if HDMI available on device via VBT */
+		for (i = 0; i < dev_priv->vbt.child_dev_num; i++) {
+			if (dev_priv->vbt.child_dev[i].common.device_type ==
+					DEVICE_TYPE_HDMI) {
+				if (I915_READ(VLV_DISPLAY_BASE + GEN4_HDMIB) &
+						SDVO_DETECTED)
+					intel_hdmi_init(dev,
+						VLV_DISPLAY_BASE +
+							GEN4_HDMIB, PORT_B);
+			}
 		}
 	} else if (SUPPORTS_DIGITAL_OUTPUTS(dev)) {
 		bool found = false;
-- 
1.7.9.5

